src/DotNetWorker.Core/FunctionMetadata/DefaultFunctionMetadata.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Collections.Generic;
+
+namespace Microsoft.Azure.Functions.Worker.Core.FunctionMetadata
+{
+    /// <summary>
+    /// Local representation of FunctionMetadata
+    /// </summary>
+    public class DefaultFunctionMetadata : IFunctionMetadata
+    {
+        /// <summary>
+        /// Creates a DefaultFunctionMetadata with the minimum required properties for successful function loading.
+        /// </summary>
+        /// <param name="functionId">Unique ID associated with each function.</param>
+        /// <param name="language">The language that the function is written in.</param>
+        /// <param name="name">The function name</param>
+        /// <param name="entryPoint">The function entrypoint</param>
+        /// <param name="rawBindings">List of the functions bindings in JSON string format</param>
+        /// <param name="scriptFile">The path to the function app assembly</param>
+        public DefaultFunctionMetadata(string functionId, string language, string name, string entryPoint, IList<string> rawBindings, string scriptFile)
```

Comment:
Is this ctor necessary if you have setters on all the properties?

I only keep constructors around if there is logic in it beyond setting (arg validation or parsing for example). In this case, you may be fine with just the default ctor and use object initializers instead.

Reply:
It isn't necessary it was a change to cut down on the lines in the source generator and the source-generated file, but at the length it is what's a bit more 😅

Reply:
If you want to keep that ctor, add a parameterless one as well.